### PR TITLE
virttest: order pci controller devices before insert

### PR DESCRIPTION
**Pull request by xutian**
add auto-increament 'slot' params when add root port

Signed-off-by: Xu Tian <xutian@redhat.com>

### CODE REVIEW

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -1293,6 +1293,32 @@ def _get_pci_bus(devices, params, dtype=None, pcie=False):
...
+                try:
+                    return order_pcics.index(dev.get_param('driver'))
+                except ValueError:
+                    return -1
+
+            pcics = []
+            for pcic in params.objects("pci_controllers"):
+                dev = devices.pcic_by_params(pcic, params.object_params(pcic))
+                slot = len([_ for _ in pcics if _.get_param('driver') == 'ioh3420'])
+                dev.set_param('slot', hex(slot))
```
Comment (vivianQizhu) [inline virttest/qemu_vm.py]: If driver is not ioh3420, slot will not be assigned value, will this line hit error therefore?

Comment (xutian) [inline virttest/qemu_vm.py]: oh, right. it's a bug. let me fix it

File: virttest/qemu_vm.py [hunk 2]
```diff
@@ -1293,6 +1293,33 @@ def _get_pci_bus(devices, params, dtype=None, pcie=False):
...
+                except ValueError:
+                    return -1
+
+            pcics = []
+            for pcic in params.objects("pci_controllers"):
+                dev = devices.pcic_by_params(pcic, params.object_params(pcic))
+                slot = len([_ for _ in pcics if _.get_param('driver') == 'ioh3420'])
+                if dev.get_param('driver') == 'ioh3420':
+                    dev.set_param('slot', hex(slot))
+                pcics.append(dev)
```
Comment (vivianQizhu) [inline virttest/qemu_vm.py]: 'slot' get value by go through the list 'pcics', but the list 'pcics' is append after that, if first loop 'pcic' has driver 'ioh3420', currently list 'pcics' is empty, I am not sure, looks slot still not assigned value in this case?

Comment (vivianQizhu) [inline virttest/qemu_vm.py]: Maybe initialize 'slot' with 0?

Comment (xutian) [inline virttest/qemu_vm.py]: right,  if it's first one, it should be slot 0.

**Comment by xutian** [comment]
@vivianQizhu  any comments and suggestion for this PR?

**Comment by jingzhao84** [comment]
PASS 1-smp_4.4096m.repeat1.test.Host_RHEL.m7.u4.q35.spice.ovmf.qcow2.virtio_scsi.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads

PASS 2-smp_4.4096m.repeat1.test.Host_RHEL.m7.u4.q35.spice.ovmf.qcow2.virtio_scsi.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.boot

cmd:
    -device ioh3420,id=root_port,slot=0x0,bus=pcie.0,addr=02 \
    -device ioh3420,id=root_port2,slot=0x1,bus=pcie.0,addr=03 \
    -device ioh3420,id=root_port3,slot=0x2,bus=pcie.0,addr=04 \

**Comment by xutian** [comment]
Merged, thanks @jingzhao84  and @vivianQizhu  comments and test results.

**Pull request merged**